### PR TITLE
Fix loss NaN bug when training HRSID dataset.

### LINKED ISSUE #759: [Bug] Cannot run Oriented R-CNN and redet on HRSID, loss NAN
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmrotate/issues) and [Discussions](https://github.com/open-mmlab/mmrotate/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmrotate.readthedocs.io/en/1.x/notes/4_faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmrotate) or [latest version (1.x)](https://github.com/open-mmlab/mmrotate/tree/dev-1.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

1.x branch https://github.com/open-mmlab/mmrotate/tree/1.x

### Environment

sys.platform: linux
Python: 3.8.16 (default, Jan 17 2023, 23:13:24) [GCC 11.2.0]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0,1,2: GeForce RTX 2080 Ti
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 9.1, V9.1.8
GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609
PyTorch: 1.7.1
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.8.2
OpenCV: 4.7.0
MMEngine: 0.5.0
MMRotate: 1.0.0rc1+5d0491c

### Reproduces the problem - code sample

For redet, I use [mmrotate](https://github.com/open-mmlab/mmrotate/tree/dev-1.x)/[configs](https://github.com/open-mmlab/mmrotate/tree/dev-1.x/configs)/[redet](https://github.com/open-mmlab/mmrotate/tree/dev-1.x/configs/redet)/redet-le90_re50_refpn_1x_dota.py
change the dataset to hrsid.
For Oriented RCNN, I use the official code of branch dev-1.x [mmrotate](https://github.com/open-mmlab/mmrotate/tree/dev-1.x)/[configs](https://github.com/open-mmlab/mmrotate/tree/dev-1.x/configs)/[oriented_rcnn](https://github.com/open-mmlab/mmrotate/tree/dev-1.x/configs/oriented_rcnn)/oriented-rcnn-le90_r50_fpn_6x_hrsid.py

But change '../_base_/schedules/schedule_6x.py' to '../_base_/schedules/schedule_1x.py'
The dataset hrsid works well on GWD, so I think the problem is not the dataset. I cannot find out what the problem is.

### Reproduces the problem - command or script

For redet, I use [mmrotate](https://github.com/open-mmlab/mmrotate/tree/dev-1.x)/[configs](https://github.com/open-mmlab/mmrotate/tree/dev-1.x/configs)/[redet](https://github.com/open-mmlab/mmrotate/tree/dev-1.x/configs/redet)/redet-le90_re50_refpn_1x_dota.py
change the dataset to hrsid.

I use the official script of I use the official code of branch dev-1.x [mmrotate](https://github.com/open-mmlab/mmrotate/tree/dev-1.x)/[configs](https://github.com/open-mmlab/mmrotate/tree/dev-1.x/configs)/[oriented_rcnn](https://github.com/open-mmlab/mmrotate/tree/dev-1.x/configs/oriented_rcnn)/oriented-rcnn-le90_r50_fpn_6x_hrsid.py

And I use the model [oriented-rcnn-le90_r50_fpn_1x_dota](https://github.com/open-mmlab/mmrotate/blob/dev-1.x/configs/oriented_rcnn/oriented-rcnn-le90_r50_fpn_1x_dota.py)

### Reproduces the problem - error message

While training, grad_norm: nan  loss: nan  loss_rpn_cls: nan  loss_rpn_bbox: nan  loss_cls: nan  acc: 100.0000  loss_bbox: nan
and evaluation
/mmrotate/evaluation/metrics/rotated_coco_metric.py - compute_metrics - 293 - The testing results of the whole dataset is empty.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by zytx121**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

close #759 

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by liuyanyi — COMMENTED** [review]
It seems 'FilterAnnotations' is under mmdet, should we add scope?

File: configs/_base_/datasets/hrsid.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
         poly2mask=False),
     dict(type='ConvertMask2BoxType', box_type='rbox'),
     dict(type='mmdet.Resize', scale=(800, 800), keep_ratio=True),
+    dict(type='FilterAnnotations', min_gt_bbox_wh=(1e-2, 1e-2)),
```
Comment (RangiLyu) [inline configs/_base_/datasets/hrsid.py:15]:
```suggestion
    dict(type='mmdet.FilterAnnotations', min_gt_bbox_wh=(1e-2, 1e-2)),
```

**Review by RangiLyu — APPROVED** [review]
